libseccomp-sys/build.rs

```diff
@@ -5,26 +5,239 @@
 
 use std::env;
 
+mod versions;
+
 const LIBSECCOMP_LIB_PATH: &str = "LIBSECCOMP_LIB_PATH";
 const LIBSECCOMP_LINK_TYPE: &str = "LIBSECCOMP_LINK_TYPE";
+const LIBSECCOMP_SRC_PATH: &str = "LIBSECCOMP_SRC_PATH";
 
 fn main() -> Result<(), Box<dyn std::error::Error>> {
+    println!("cargo:rerun-if-env-changed={}", LIBSECCOMP_SRC_PATH);
     println!("cargo:rerun-if-env-changed={}", LIBSECCOMP_LIB_PATH);
     println!("cargo:rerun-if-env-changed={}", LIBSECCOMP_LINK_TYPE);
 
-    if let Ok(path) = env::var(LIBSECCOMP_LIB_PATH) {
-        println!("cargo:rustc-link-search=native={}", path);
+    // We do cfg on blocks here instead of `if cfg!()...` so we don't have to build the cc library
+    // at all in the non-bundled case.
+
+    // If bundled feature is enabled, build the library and return any errors.
+    #[cfg(feature = "bundled")]
+    {
```

Comment:
I think this bundled feature should be fired only when the target is musl, so it would be better to output error message if the target is not musl. What do you think?
@boustrophedon @rusty-snake

Reply:
It's entirely possible to use the bundled feature with glibc:

```
[hcs libseccomp]$ cargo test --features bundled -- --test-threads=1
[...]
[hcs libseccomp]$ ldd ../target/debug/deps/libseccomp-7f04e6b794ba92eb
	linux-vdso.so.1 (0x00007ffe2798c000)
	libgcc_s.so.1 => /usr/lib/libgcc_s.so.1 (0x00007faf6eeb9000)
	libm.so.6 => /usr/lib/libm.so.6 (0x00007faf6edd1000)
	libc.so.6 => /usr/lib/libc.so.6 (0x00007faf6ebc7000)
	/lib64/ld-linux-x86-64.so.2 => /usr/lib64/ld-linux-x86-64.so.2 (0x00007faf6f023000)
```

For example, if you are writing an application where you want to statically link libseccomp but dynamically link a proprietary dependency that requires glibc (and then use seccomp to make sure the dependency isn't doing anything bad!), it would be very hard to make everything work properly.